ledgerjs-fio/src/fio.ts

```diff
@@ -0,0 +1,335 @@
+        this.transport = transport
+        // Note: this is list of methods that should "lock" the transport to avoid concurrent use
+        const methods = [
+            "getVersion",
+            "getSerial",
+            "getPublicKey",
+            "signTransaction",
+        ]
+        this.transport.decorateAppAPIMethods(this, methods, scrambleKey)
+        this._send = async (params: SendParams): Promise<Buffer> => {
+            let response = await wrapConvertDeviceStatusError(this.transport.send)(
+                CLA,
+                params.ins,
+                params.p1,
+                params.p2,
+                params.data,
+            )
+            response = utils.stripRetcodeFromResponse(response)
+
+            if (params.expectedResponseLength != null) {
+                assert(
+                    response.length === params.expectedResponseLength,
+                    `unexpected response length: ${response.length} instead of ${params.expectedResponseLength}`,
+                )
+            }
+
+            return response
+        }
+    }
+
+    /**
+     * Returns an object containing the app version.
+     *
+     * @returns Result object containing the application version number.
+     *
+     * @example
+     * const { major, minor, patch, flags } = await fio.getVersion();
+     * console.log(`App version ${major}.${minor}.${patch}`);
+     *
+     */
+    async getVersion(): Promise<GetVersionResponse> {
+        const version = await interact(this._getVersion(), this._send)
+        return {version, compatibility: getCompatibility(version)}
+    }
+
+    /** @ignore */
+    * _getVersion(): Interaction<Version> {
+        return yield* getVersion()
+    }
+
+    /**
+     * Returns an object containing the device serial number.
+     *
+     * @returns Result object containing the device serial number.
+     *
+     * @example
+     * const { serial } = await fio.getSerial();
+     * console.log(`Serial number ${serial}`);
+     *
+     */
+    async getSerial(): Promise<GetSerialResponse> {
+        return interact(this._getSerial(), this._send)
+    }
+
+    /** @ignore */
+    * _getSerial(): Interaction<GetSerialResponse> {
+        const version = yield* getVersion()
+        return yield* getSerial(version)
+    }
+
+    /**
+     * Get public key for the specified BIP 32 path.
+     *
+     * @param path The path. A path must begin with `44'/235'/0'/0/i`.
+     * @returns The public key (i.e. with chaincode).
+     *
+     * @example
+     * ```
+     * const publicKey = await fio.getPublicKey[[ HARDENED + 44, HARDENED + 235, HARDENED + 0, 0, 0 ]);
+     * console.log(publicKey);
+     * ```
+     */
+    async getPublicKey(
+        {path, show_or_not}: GetPublicKeyRequest
+    ): Promise<GetPublicKeyResponse> {
+        // validate the input
+        validate(isValidPath(path), InvalidDataReason.GET_EXT_PUB_KEY_PATHS_NOT_ARRAY)
```

Comment:
In this step as indicated also by the `InvalidDataReason` you are just checking whether the input is an array. Parsing of an actual path is done in the next step.
```suggestion
        validate(isArray(path), InvalidDataReason.GET_EXT_PUB_KEY_PATHS_NOT_ARRAY)
```